docs/proposals/20230612-node-os-image-caching.md

```diff
@@ -95,51 +96,94 @@ We will know we’ve succeeded when we can benchmark speed increases and success
+  - lastTransitionTime: "2023-06-12T23:14:55Z"
+    status: "True"
+    type: ScaleSetDesiredReplicas
+  - lastTransitionTime: "2023-06-12T23:14:55Z"
+    status: "True"
+    type: ScaleSetModelUpdated
+  - lastTransitionTime: "2023-06-12T23:14:55Z"
+    status: "True"
+    type: ScaleSetRunning
+```
 
-As for why the healthy node has to be shut down while creating a snapshot of it, if it isn’t shut down first then pods can be scheduled as the snapshot is taken which will cause some dangerous states in terms of how it exists after being utilized by the AzureMachineTemplates.
+Example AzureMachine yaml with the proposed additional needed timestamps in status:
 
-In terms of how a healthy node would be selected, there is already an annotation on each node from its status as to when they’ve last been patched. An ideal node would be one which has been patched since the last prototype went into service and is running and healthy. Whichever node has been steadily healthy for the longest amount of time since the last patch and has the patch applied should be chosen as it’s the most overall stable. As the prototype is always from a successfully healthy and working node the image is always known to be working before being chosen for replication.
+```yaml
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: AzureMachine
+metadata:
+  name: node-os-image-caching-machine
+  namespace: default
+status:
+  conditions:
+  - lastTransitionTime: "2023-06-12T23:14:55Z"
+    status: "True"
+    type: ScaleSetModelUpdated
+  - lastTransitionTime: "2023-06-12T23:14:55Z"
+    status: "True"
+    type: ScaleSetRunning
+```
 
 In terms of when to take a snapshot, a day is given as a general example which should be good for typical use but the specification of how often will be customizable as we know that certain operators have different strategies and use cases for how they’re running their services on our clusters.
 
-In terms of data model changes, AzureMachine and AzureMachinePool changes are likely and the changes we expect will be purely additive and nonbreaking. No removals should be required to the data model. For AzureMachine on AzureMachineTemplate we will add a new optional field under spec.template.spec called nodeCachingMode which will be enabled if present and a sub field under that called cacheInterval which will be set to 1 day by default, and it will be a required subfield of nodeCachingMode if that field is there. For AzureMachinePool we will do the same thing under spec.template in the AzureMachinePool files.
+In terms of data model changes, AzureMachine and AzureMachinePool changes are likely and the changes we expect will be purely additive and nonbreaking. No removals should be required to the data model. For AzureMachineTemplate and AzureMachinePool we will add a new optional field under meta.annotations called nodeCachingModeInterval which will be enabled if present and it will map to an interval of 1 day by default.
```

Comment:
> In terms of data model changes, AzureMachine and AzureMachinePool changes are likely and the changes we expect will be purely additive and nonbreaking.

I think the little bit of conjecture here was ok for the initial issue before drafting the proposal, but I think you have enough here now that this can be a bit more assertive.